cfgov/v1/models/base.py

```diff
@@ -181,12 +181,19 @@ def specific_categories_query(block, parent_slug):
                     block, 'archive-past-events') & query
             relate = block.value.get('relate_{}'.format(search_type), None)
             if relate:
-                related[search_type_name] = (
+                type_query = (
                     AbstractFilterPage.objects.live().filter(
                         search_query
                     ).distinct().exclude(id=self.id).order_by(
                         '-date_published'
-                    )[:block.value['limit']])
+                    )
+                )
+                # Apply similar logic as snippets.py's filter_by_tags method
+                # to enable AND filtering
+                if block.value['and_filtering']:
+                    for tag in self.tags.names() or []:
```

Comment:
@richaagarwal I think this might be a copy of what @Scotchester did [here](https://github.com/cfpb/cfgov-refresh/blob/master/cfgov/v1/models/snippets.py#L75), which is similar - in that case, because the `filter_by_tags` was being implemented in a separate method, we added the `or []` to support the case where `None` was passed in. You're right that in this case, it's unnecessary.